code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -0,0 +1,210 @@
+	if(GLOB.reality_smash_track == src)
+		stack_trace("/datum/reality_smash_tracker was deleted. Heretics may no longer access any influences. Fix it or call coder support")
+	for(var/X in smashes)
+		qdel(X)
+	for(var/Y in targets)
+		targets -= Y
+	. = ..()
+
+/**
+  * Automatically fixes the target and smash network
+  *
+  * Fixes any bugs that are caused by late Generate()
+  */
+/datum/reality_smash_tracker/proc/ReworkNetwork()
+	for(var/mind in targets)
+		for(var/X in smashes)
+			var/obj/effect/reality_smash/reality_smash = X
+			reality_smash.AddMind(mind)
+
+/**
+  * Generates a set amount of reality smashes based on the N value
+  *
+  * Automatically creates more reality smashes
+  */
+/datum/reality_smash_tracker/proc/Generate(var/n)
+	var/number = round(max(n * 6,10),1)
+
+	for(var/i in 0 to number)
+
+		var/obj/effect/landmark/L = pick(GLOB.generic_event_spawns + GLOB.landmarks_list)
+		var/obj/effect/reality_smash/RS = new(L.drop_location())
+		smashes += RS
+
+	ReworkNetwork()
+
+/**
+  * Adds a mind to the list of people that can see the reality smashes
+  *
+  * Use this whenever you want to add someone to the list
+  */
+/datum/reality_smash_tracker/proc/AddMind(var/datum/mind/M)
+	targets |= M
+	for(var/X in smashes)
+		var/obj/effect/reality_smash/reality_smash = X
+		reality_smash.AddMind(M)
+
+/**
+  * Removes a mind from the list of people that can see the reality smashes
+  *
+  * Use this whenever you want to remove someone from the list
+  */
+/datum/reality_smash_tracker/proc/RemoveMind(var/datum/mind/M)
+	targets -= M
+	for(var/obj/effect/reality_smash/RS in smashes)
+		RS.RemoveMind(M)
+
+/obj/effect/broken_illusion
+	name = "Pierced reality"
+	icon = 'icons/effects/eldritch.dmi'
+	icon_state = "pierced_illusion"
+
+/obj/effect/broken_illusion/examine(mob/user)
+	if(!IS_HERETIC(user) && ishuman(user))
+		var/mob/living/carbon/human/human_user = user
+		to_chat(human_user,"<span class='warning'>Your brain hurts when you look at this!</span>")
+		human_user.adjustOrganLoss(ORGAN_SLOT_BRAIN,10)
+	. = ..()
+
+/obj/effect/reality_smash
+	name = "/improper reality smash"
+	icon = 'icons/effects/eldritch.dmi'
+	///We cannot use icon_state since this is invisible, functions the same way but with custom behaviour.
+	var/image_state = "reality_smash"
+	///Who can see us?
+	var/list/minds = list()
+	///Tracked image
+	var/image/img
+
+/obj/effect/reality_smash/Initialize()
+	. = ..()
+	img = image(icon, src, image_state, OBJ_LAYER)
+	generate_name()
+
+/obj/effect/reality_smash/Destroy()
+
+	for(var/datum/mind/cultie in minds)
+		if(cultie && cultie.current && cultie.current.client)
+			cultie.current.client.images -= img
+		//clear the list
+		minds -= cultie
+	img = null
+	new /obj/effect/broken_illusion(drop_location())
+	return ..()
+
+///Makes the mind able to see this effect
+/obj/effect/reality_smash/proc/AddMind(var/datum/mind/cultie)
+	if(cultie.current.client)
+		minds |= cultie
```

Comment:
1) May `AddMind` be called on someone already in the list, warranting the `|=` ?
2) What happens if they disconnect and connect later? Will they be missing from the list?

Reply:
Yes, it can happen when Admin is generating influences post-roundstart, in which case ReworkNetwork() on reality_smash_tracker is called that readds any heretics that were no on the list. That's why it is |= so that it isn't added twice,

Reply:
and no they wont be mising from the list